Fix - Cannot enter address in personal details, address disappears after entering #28474

AmjedNazzal opened the pull request:
### Details
We are adding a street fallback for instances where neither street_number nor route exist in the API response

### Fixed Issues

$ https://github.com/Expensify/App/issues/28474
PROPOSAL: https://github.com/Expensify/App/issues/28474#issuecomment-1783289406

### Tests

1. Go to Settings > Profile > Personal details > Address
2. In address line 1, type "200 dury" or "161 sajik" and pick the first result
3. Observe that  address line 1 gets filled correctly with the choosen address

- [x] Verify that no errors appear in the JS console

### Offline tests

No test steps as google autocomplete doesn't work offline

### QA Steps

1. Go to Settings > Profile > Personal details > Address
2. In address line 1, type "200 dury" or "161 sajik" and pick the first result
3. Observe that  address line 1 gets filled correctly with the choosen address

- [x] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
        - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/74202751/13eab261-0b86-4e62-857b-e2610b5c2a2d

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/74202751/1e172f66-0c5e-4d60-a156-bc8fb9d6423e

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/74202751/20faa7cb-9d62-4a7f-8b7d-7b9bc0371b3f

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/74202751/5d099be7-54e6-4530-8a84-56f5212a7042

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/74202751/a5654580-fa8d-4812-81f4-965ef5783898

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/74202751/522fb29d-e567-4770-a14d-30072f03df02

</details>

AmjedNazzal committed:
Fix Issue #28474

AmjedNazzal committed:
Merge branch 'Expensify:main' into Issue28474

AmjedNazzal committed:
Merge branch 'Expensify:main' into Issue28474

AmjedNazzal committed:
Fix Issue #28474 small change

jjcoffee commented on src/components/AddressSearch/index.js:
```diff
@@ -254,6 +255,30 @@ function AddressSearch(props) {
+            const {adr_address} = details;
+            const streetAddressRegex = /(<span\s+class\s*=\s*"street-address"[^>]*>[^<]*<\/span>)/;
+            const streetAddressHtml = adr_address.match(streetAddressRegex);
+            if (streetAddressHtml[1]) {
```
This will be undefined if there's no match

AmjedNazzal replied:
This is why I kept the else code, but supposidly google API will always return adr_address in that format

jjcoffee replied:
I mean it will throw an error, since if it fails `match` will return null and not an array.

AmjedNazzal replied:
Yeah you're right, I'll add a catch of some sort or change something else here soon

jjcoffee replied:
Sure, no rush - I just wanted to give it a quick look over. I don't have time to run all the tests today anyway :smile:

jjcoffee commented on src/components/AddressSearch/index.js:
```diff
@@ -254,6 +255,30 @@ function AddressSearch(props) {
+                const parser = new ExpensiMark();
+                const streetFallback = parser.htmlToText(streetAddressHtml[1]);
+                values.street = streetFallback;
+            } else {
```
I think we can just leave this part out for this PR, it feels a bit too hacky to me.

AmjedNazzal replied:
Do you mean not use adr_address or the parser itself?

jjcoffee replied:
Ooh sorry the highlight didn't work, I meant the fallback to the fallback isn't necessary.

AmjedNazzal replied:
Ahhhh got you, Okay I'll just focus on ad_address then

AmjedNazzal committed:
Merge branch 'Expensify:main' into Issue28474

AmjedNazzal committed:
Fix issue #28474 removed the extra fallback

AmjedNazzal commented:
@jjcoffee I've made some changes, I've removed the extra fallback and I've implemented `_.get ` to avoid potential error throwing, also added an extra check for `details.adr_address` to ensure that we don't try calling .match on an undefined prop. let me know what you think :)

AmjedNazzal committed:
Merge branch 'Expensify:main' into Issue28474

AmjedNazzal committed:
Fix issue #28474 reverted to plain regex

AmjedNazzal commented:
Sorry that I keep making changes but what I decided is that I'm already using a regex to capture the street address so there is no point in using ExpensiMark here since it will be passed plain text string anyway and it uses regex in a similar way as well regardless so I don't see the benefit of using it, I think just simple plain regex should suffice here, otherwise we could make a custom parser or use a parsing library, which looking at our package.json it seem like we already have htmlparser2 v7.2.0, although we don't seem to be using it anywhere in the app so not sure why it's there.

AmjedNazzal committed:
Merge branch 'Expensify:main' into Issue28474

AmjedNazzal committed:
Merge branch 'Expensify:main' into Issue28474

AmjedNazzal commented:
@jjcoffee Thoughts?

jjcoffee commented:
I think I prefer parsing vs regex, and it looks like htmlparser2 is used in a few places. Not a huge deal if it's too much of a pain to implement but it might be worth a try!

AmjedNazzal commented:
@jjcoffee It's not a pain at all! here's my code to implement htmlparser2 for this, let me know what you think of it, until then I will be doing some testing to see if there seem to be any potential issues with using this code

```javascript
if (!values.street && details.adr_address) {
  let streetAddress;
  let matchedText = false;

  const parser = new HtmlParser({
      onopentag(name, attribs){
          if(name === "span" && attribs.class === "street-address"){
              matchedText = true;
          }
      },
      ontext(text){
          if(matchedText){
              streetAddress = text;
          }
      },
      onclosetag(tagname){
          if(tagname === "span"){
              matchedText = false;
          }
      }
  }, { decodeEntities: true });
  parser.write(details.adr_address);
  parser.end();

  if (streetAddress) {
      values.street = streetAddress.trim();
  } 
}
```

jjcoffee commented on src/components/AddressSearch/index.js:
```diff
@@ -254,6 +254,17 @@ function AddressSearch(props) {
+        if (!values.street && details.adr_address) {
+            const streetAddressRegex = /<span\s+class\s*=\s*"street-address"[^>]*>([^<]*)<\/span>/;
+            const adr_address = details.adr_address.match(streetAddressRegex);
+            const streetAddressFallback = _.get(adr_address, [1], null);
```
Any reason why we can't use `lodashGet` here instead?

AmjedNazzal replied:
I'm already using _.get, is this an old comment I didn't see before?

jjcoffee replied:
I'm talking about `lodashGet` e.g. here

https://github.com/Expensify/App/blob/b81148810c83bba7de2dcdd22dc4ac9878b0b9c5/src/components/AddressSearch/index.js#L240

jjcoffee replied:
It's not a big deal since we're all using underscores in this file, but it's nicer to use it when we can!

AmjedNazzal replied:
In that case then no there is no specific reason I used _, I pretty much did only because it's already imported and used all over the component so it doesn't really make a difference from what I know.

jjcoffee replied:
Let's switch it to `lodashGet` as we prefer it in general. Makes it easier if we want to get rid of underscores altogether in future!

AmjedNazzal replied:
Done!

jjcoffee commented:
Ah this doesn't look as tidy as I imagined! I think it might be overkill for this case and we can probably stick with the regex for now, unless you don't want your extra work to go to waste :smile: We can also see what the internal engineer thinks. 

I'll try to get the checklist done tomorrow, unfortunately higher priority PRs got in the way today!

AmjedNazzal commented:
@jjcoffee Haha no don't worry about it, I don't mind trying different things to see what feels right, we can certently stick to the regex way, it should work fine either way since we already use similar methods in ExpensiMark anyway :)

jjcoffee commented on src/components/AddressSearch/index.js:
```diff
@@ -254,6 +254,17 @@ function AddressSearch(props) {
+        // Some edge-case addresses may lack both street_number and route in the API response, resulting in an empty "values.street"
+        // We are setting up a fallback to ensure "values.street" is populated with a relevant value
+        if (!values.street && details.adr_address) {
+            const streetAddressRegex = /<span\s+class\s*=\s*"street-address"[^>]*>([^<]*)<\/span>/;
```
```suggestion
            const streetAddressRegex = /<span class="street-address">([^<]*)<\/span>/;
```
I've not tested this yet, but I wonder how necessary all these extra whitespace checks are. Since we've agreed this will follow a standardised format, I think we can do a straight check for the exact span with this specific class.

AmjedNazzal replied:
I just thought it wouldn't hurt to be extra careful but yeah sure since it's a standardised format then the white space checks are not a must or anything.

jjcoffee commented:
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Android: mWeb Chrome</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS: Native</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS: mWeb Safari</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>MacOS: Desktop</summary>

<!-- add screenshots or videos here -->

</details>

AmjedNazzal committed:
Update src/components/AddressSearch/index.js

Co-authored-by: Joel Davies <joeld.dev@gmail.com>

AmjedNazzal committed:
Fix Issue #28474 switched to lodashget

jjcoffee commented:
@AmjedNazzal Testing on a distance request, I'm getting a bit of a weird result where only the street number displays:

https://github.com/Expensify/App/assets/27287420/5073a98a-fb8d-4925-931f-68d5418a10f2

AmjedNazzal commented:
@jjcoffee That's due to how `DistanceRequest` sets up it's waypoints and it's waypoints preview, it uses values.address which holds the full address like this "200 Duryugongwon-ro, Dalseo-gu, Daegu, South Korea" and then extracts from it, it doesn't use values.street at all.

jjcoffee commented:
Ooh okay yeah I see that looks like it would be out of scope for this issue I think!

jjcoffee reviewed (approved):
LGTM!

jjcoffee commented:
@cristipaval All yours! Just a quick note that we did look at parsing the HTML instead, but it ended up not being as [elegant](https://github.com/Expensify/App/pull/30819#issuecomment-1799342627) as I had hoped, so I decided it's not worth it for this when a very simple regex check will do.

There is also [this issue](https://github.com/Expensify/App/pull/30819#issuecomment-1803986084) for distance requests, but it seems out of scope as distance requests do some [manual parsing](https://github.com/Expensify/App/pull/30819#issuecomment-1804038113) of the address for display.

AmjedNazzal committed:
Merge branch 'Expensify:main' into Issue28474

AmjedNazzal committed:
Merge branch 'Expensify:main' into Issue28474

AmjedNazzal committed:
Merge branch 'Expensify:main' into Issue28474

AmjedNazzal committed:
Merge branch 'Expensify:main' into Issue28474

cristipaval commented:
> @cristipaval All yours! Just a quick note that we did look at parsing the HTML instead, but it ended up not being as [elegant](https://github.com/Expensify/App/pull/30819#issuecomment-1799342627) as I had hoped, so I decided it's not worth it for this when a very simple regex check will do.
> 
> There is also [this issue](https://github.com/Expensify/App/pull/30819#issuecomment-1803986084) for distance requests, but it seems out of scope as distance requests do some [manual parsing](https://github.com/Expensify/App/pull/30819#issuecomment-1804038113) of the address for display.

Agree! Thank you both!

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/6869840079) to staging by https://github.com/cristipaval in version: 1.3.99-0 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/6882673474) to production by https://github.com/roryabraham in version: 1.3.99-0 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/6882677947) to staging by https://github.com/cristipaval in version: 1.4.0-0 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅
